src/gui/applicationpalette.h

```diff
@@ -0,0 +1,119 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If
+ * you modify file(s), you may extend this exception to your version of the
+ * file(s), but you are not obligated to do so. If you do not wish to do so,
+ * delete this exception statement from your version.
+ */
+ 
+#include <QColor>
+#include <QPalette>
+#include <QWidget>
+
+class ApplicationPalette : public QWidget
+{
+    Q_OBJECT
+
+    Q_PROPERTY(QColor window READ window WRITE setWindow)
+    Q_PROPERTY(QColor windowText READ windowText WRITE setWindowText)
+    Q_PROPERTY(QColor windowTextDisabled READ windowTextDisabled WRITE setWindowTextDisabled)
+    Q_PROPERTY(QColor base READ base WRITE setBase)
+    Q_PROPERTY(QColor alternateBase READ alternateBase WRITE setAlternateBase)
+    Q_PROPERTY(QColor text READ text WRITE setText)
+    Q_PROPERTY(QColor textDisabled READ textDisabled WRITE setTextDisabled)
+    Q_PROPERTY(QColor toolTipBase READ toolTipBase WRITE setToolTipBase)
+    Q_PROPERTY(QColor toolTipText READ toolTipText WRITE setToolTipText)
+    Q_PROPERTY(QColor toolTipTextDisabled READ toolTipTextDisabled WRITE setToolTipTextDisabled)
+    Q_PROPERTY(QColor brightText READ brightText WRITE setBrightText)
+    Q_PROPERTY(QColor brightTextDisabled READ brightTextDisabled WRITE setBrightTextDisabled)
+    Q_PROPERTY(QColor highlight READ highlight WRITE setHighlight)
+    Q_PROPERTY(QColor highlightedText READ highlightedText WRITE setHighlightedText)
+    Q_PROPERTY(QColor highlightedTextDisabled READ highlightedTextDisabled WRITE setHighlightedTextDisabled)
+    Q_PROPERTY(QColor button READ button WRITE setButton)
+    Q_PROPERTY(QColor buttonText READ buttonText WRITE setButtonText)
+    Q_PROPERTY(QColor buttonTextDisabled READ buttonTextDisabled WRITE setButtonTextDisabled)
+    Q_PROPERTY(QColor link READ link WRITE setLink)
+    Q_PROPERTY(QColor linkVisited READ linkVisited WRITE setLinkVisited)
+    Q_PROPERTY(QColor light READ light WRITE setLight)
+    Q_PROPERTY(QColor midlight READ midlight WRITE setMidlight)
+    Q_PROPERTY(QColor mid READ mid WRITE setMid)
+    Q_PROPERTY(QColor dark READ dark WRITE setDark)
+    Q_PROPERTY(QColor shadow READ shadow WRITE setShadow)
+
+public:
+    ApplicationPalette(QWidget *parent = nullptr) : QWidget(parent) {}
+
+private:
+    QColor window();
```

Comment:
`QColor window() const;` and this applies to other getter functions too.

Reply:
done.